src/main/scala/edu/ie3/simona/agent/participant/evcs/EvcsAgentFundamentals.scala

```diff
@@ -307,48 +307,44 @@ protected trait EvcsAgentFundamentals
     )
   }
 
-  /** Handles a evcs movements message that contains information on arriving and
-    * departing vehicles. After applying the movements to the last known set of
-    * parked evs, calculates resulting charging power. Sends completion message
-    * to scheduler without scheduling new activations.
-    * @param currentTick
-    *   The current tick that has been triggered
-    * @param scheduler
-    *   The scheduler ref
+  /** Handle a request for returning those EVs that are departing at the current
+    * tick. SOC and results are calculated and corresponding listeners are
+    * informed.
+    *
+    * @param tick
+    *   The current simulation tick
     * @param modelBaseStateData
-    *   The state data
-    * @param evcsMovementsData
-    *   The movement data that has been received
-    * @return
-    *   [[Idle]] with updated result values
+    *   The current Base state data
+    * @param requestedDepartingEvs
+    *   The UUIDs of EVs that are requested to be returned
     */
-  private def handleEvMovementsAndGoIdle(
-      currentTick: Long,
-      scheduler: ActorRef,
+  protected def handleDepartingEvsRequest(
+      tick: Long,
       modelBaseStateData: ParticipantModelBaseStateData[
-        _ <: ApparentPower,
-        _,
-        _
+        ApparentPower,
+        EvcsRelevantData,
+        EvcsModel
       ],
-      evcsMovementsData: EvMovementData
-  ): FSM.State[AgentState, ParticipantStateData[ApparentPower]] = {
+      requestedDepartingEvs: Seq[UUID]
+  ): ParticipantModelBaseStateData[
+    ApparentPower,
+    EvcsRelevantData,
+    EvcsModel
+  ] = {
+
     val evServiceRef = getService[ActorEvMovementsService](
       modelBaseStateData.services
     )
 
     val (tickInterval, lastEvs) =
-      getTickIntervalAndLastEvs(currentTick, modelBaseStateData)
+      getTickIntervalAndLastEvs(tick, modelBaseStateData)
+
+    validateDepartures(lastEvs, requestedDepartingEvs)
```

Comment:
I wouldn't agree to that since the EVs retrieved here stem from a tick in the past, thus `lastEvs`. This is in contrast to the EVs that are determined for the current tick, namely the ones staying plus arrivals, thus `currentEvs`.

Reply:
This also makes sense if you look at the naming of `currentTick`, which is the tick currently activated. A tick at which the agent was active _before the current tick_ I would name `lastTick` :)

Reply:
I see the point (and was expecting that you will make it 😄 ) but I would understand it a bit differently.
Since we haven't dealt with arrivals and departures yet, the current evs are still the evs from last tick. Nevertheless I don't think we have to drag that on here.